core/cloudflow-akka-tests/src/test/java/cloudflow/akkastream/javadsl/AkkaStreamletTest.java

```diff
@@ -117,6 +157,48 @@ public void run() {
     }
   }
 
+  class FileWritingProcessor extends AkkaStreamlet {
+
+    private final String volumeMountName;
+
+    FileWritingProcessor(String volumeMountName) {
+      this.volumeMountName = volumeMountName;
+    }
+
+    AvroInlet<Data> inlet = AvroInlet.<Data>create("in", Data.class);
+    AvroOutlet<Data> outlet = AvroOutlet.<Data>create("out", Data.class);
+
+    public StreamletShape shape() {
+      return StreamletShape.createWithInlets(inlet).withOutlets(outlet);
+    }
+
+    @Override
+    public VolumeMount[] defineVolumeMounts() {
+      return new VolumeMount[] {VolumeMount.createReadWriteMany(volumeMountName, "path")};
+    }
+
+    public AkkaStreamletLogic createLogic() {
+      Path mountedPath = getContext().getMountedPath(defineVolumeMounts()[0]);
+      return new AkkaStreamletLogic(getContext()) {
+        public void run() {
+          getSourceWithOffsetContext(inlet)
+              .via(
+                  Flow.<Pair<Data, CommittableOffset>>create()
```

Comment:
I was going to recommend using Akka's `FileIO` and its [`toPath`](https://doc.akka.io/docs/akka/current/stream/operators/FileIO/toPath.html) sink but then I realized that that would lead to two sinks. Too bad FileIO doesn't offer Flow versions of those methods.

Reply:
Yeah not ideal. The best that I could come up with to use `FileIO` is the following:
```getSourceWithCommittableContext(inlet)
                  .asSource()
                  .via(Flow.<Pair<Data, ConsumerMessage.Committable>>create().alsoTo(getCommittableSink(outlet)))
                  .map(dataIn -> ByteString.fromString(dataIn.first().toString()))
                  .to(FileIO.toPath(mountedPath))
                  .run(materializer());```

Unfortunately Java poor type inference didn't allow me to do any better, but if you prefer it I could check this in. WDYT @agemooij?

Reply:
No, unfortunately that would not work as you would expect since the reads would be committed before/during/after the elements were actually written to file, negating the whole point of "commit-reads-after-writes".

So without a flow-based version of FileIO, your original code is the best solution since it properly orders the operations.